### ISSUE TITLE
Replace `LogEntry` Rust struct with a protobuf object

### ISSUE THREAD

**Issue by tiziano88** [project-oak/oak#568]
https://github.com/project-oak/oak/blob/1528e03883bf23e727f858e1342f335ddf041b06/sdk/rust/oak_log/src/lib.rs#L31-L37

Currently `LogEntry` is a Rust struct wrapping a string, and on the C++ side of the logging pseudo-node, this is de-serialized as a string, but there is no "schema" to keep the two sides in sync, especially if we wanted to add more fields to it (e.g. log level).

@daviddrysdale : it looks like protobuf is going to be the _de facto_ IDL, at least for the time being, so perhaps we should replace this object (and others) with corresponding protobuf ones. What are your thoughts?

**Comment by daviddrysdale** [comment]
Yeah, I think that makes sense for now.  Something like the following?
```proto
message LogMessage {
  string file = 1;
  int32 line = 2;
  int32 level = 3;
  string message = 4;
}
```

**Comment by tiziano88** [comment]
For reference, this is the corresponding Rust struct: https://docs.rs/log/0.4.10/log/struct.Record.html (not that we have to necessarily stick to that though).

**Comment by tiziano88** [comment]
Assigning to @rbehjati to investigate, I think it makes sense to go ahead with that proto definition, which perhaps should live near or inside `oak_abi` (especially after #666 ).

FYI the `log` crate is in the process of adding key/value support (see https://docs.rs/log/0.4.10/log/kv/index.html ), so we may look into supporting that in the future.

**Comment by rbehjati** [comment]
The change list in (#697) relies on `rust-protobuf` for generating rust structs from proto file. The use of `rust-protobuf` in the entire crate will later be replaced with `prost` in a separate task.
